bot/src/main/java/org/openjdk/skara/bot/BotRunner.java

```diff
@@ -60,6 +59,29 @@ enum TaskPhases {
 
     private final AtomicInteger workIdCounter = new AtomicInteger();
 
+    /**
+     * A wrapper for a WorkItem while it's tracked as pending. Used to track
+     * when a particular WorkItem entered the pending state so that metrics
+     * and log messages can use this information.
+     */
+    private static class PendingWorkItem {
+        private final WorkItem item;
```

Comment:
I actually think this is completely fine. The private fields can only be accessed within this java source file, so there is no risk of messing things up for unknown clients in the future. Adding extra boilerplate seems unnecessary to me. The fields should definitely not be public IMO.

Maybe this would qualify as a record, but I don't think we have started using them in Skara yet. I'm also not sure I want the automatically generated equals and hashcode from a record on this class. It works well with the default Object.hashcode().